scratchapi2/user.py

```diff
@@ -138,96 +161,81 @@ def info(self):
         self.about_me = self.status
         self.what_im_working_on = self.bio
         self.what_working_on = self.bio
-        return self.__dict__
+        return self.__dict__.copy()
 
     def following(self, limit=100, offset=0):
-        """ Return whom the user follows. limit and offset available. """
-        try:
-            req = requests.get("https://api.scratch.mit.edu/users/{0}/following?limit={1}&offset={2}".format(self.username, limit, offset)).json()
-        except:
-            raise ScratchAPIError
-        following_users = []
+        """Yield other Users this User follows."""
+        req = _request("users/{0}/following?limit={1}&offset={2}",
+                       self.username, limit, offset)
         for user in req:
-            following_users.append(User(user["username"], getinfo=False))
-        return following_users
+            yield User(user["username"], getinfo=False)
 
     def followers(self, limit=100, offset=0):
-        """ Return whom the user is followed by. limit and offset available. """
-        try:
-            req = requests.get("https://api.scratch.mit.edu/users/{0}/followers?limit={1}&offset={2}".format(self.username, limit, offset)).json()
-        except:
-            raise ScratchAPIError
-        followers = []
+        """Yield other Users that follow this User."""
+        req = requests.get("users/{0}/followers?limit={1}&offset={2}",
+                           self.username, limit, offset)
         for user in req:
-            followers.append(User(user["username"], getinfo=False))
-        return followers
+            yield User(user["username"], getinfo=False)
 
     def unread_messages(self):
-        """ Return the number of unread messages """
-        try:
-            req = requests.get("https://api.scratch.mit.edu/users/{0}/messages/count".format(self.username)).json()
-        except:
-            raise ScratchAPIError
+        """Return the number of messages this User has not read."""
+        req = _request("users/{0}/messages/count",
+                       self.username)
         return req["count"]
 
     def projects(self, limit=10, offset=0):
-        """ Return the user's projects. limit and offset available. """
-        try:
-            req = requests.get("https://api.scratch.mit.edu/users/{0}/projects?limit={1}&offset={2}".format(self.username, limit, offset)).json()
-        except:
-            raise ScratchAPIError
-        projects = []
+        """Yield the user's Projects."""

```

Comment:
"Projects" is capitalized to indicate it returns `Project` instances, not just the user's projects.